source/2_platform/rex_windows/include/rex_windows/console_application.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "rex_engine/core_application.h"
+#include "rex_std/utility.h"
+#include "rex_windows/platform_creation_params.h"
+
+namespace rex
+{
+  namespace win32
+  {
+    class ConsoleApplication : public CoreApplication
+    {
+    public:
+      explicit ConsoleApplication(ApplicationCreationParams&& appParams);
+      ConsoleApplication(const ConsoleApplication&) = delete;
+      ConsoleApplication(ConsoleApplication&&)      = delete;
+      ~ConsoleApplication() override;
+
+      ConsoleApplication& operator=(const ConsoleApplication&) = delete;
+      ConsoleApplication& operator=(ConsoleApplication&&)      = delete;
+
+    private:
+      bool platform_init() override;
+      void platform_update() override;
+      void platform_shutdown() override;
+
+    private:
+      class Internal;
+      static constexpr card32 s_internal_size      = 264;
+      static constexpr card32 s_internal_alignment = 8;
+      rsl::aligned_storage<s_internal_size, s_internal_alignment> m_internal_obj;
+      Internal* m_internal;
```

Comment:
All of this is to have a PIMPL structure ... 
I would still go for a rsl::unique_ptr and be done with it, an application is only created once.
If this was a hot path and we would have a lot of these objects I would understand why you would do this but this seems overkill

Reply:
this is fixed in the console app PR